### ISSUE TITLE
StandardInput instance has no attribute 'encoding'

### ISSUE THREAD

**Issue by cronicryo** [cannatag/ldap3#386]
using this package in a 3d application where the python version 2.7.6

i get the error:  StandardInput instance has no attribute 'encoding'
**ldap3/utils/config.py:94**

ive modified the if statement as follows

#### FROM:

```python
if stdin and stdin.encoding:
    _DEFAULT_ENCODING = stdin.encoding
elif getdefaultencoding():
    _DEFAULT_ENCODING = getdefaultencoding()
else:
    _DEFAULT_ENCODING = 'utf-8'
```

#### TO:

```python
if stdin and (stdin.encoding if hasattr(stdin, "encoding") else False):
    _DEFAULT_ENCODING = stdin.encoding
elif getdefaultencoding():
    _DEFAULT_ENCODING = getdefaultencoding()
else:
    _DEFAULT_ENCODING = 'utf-8'
```

**Comment by cannatag** [comment]
I've slightly modified your code: 

```
if stdin and hasattr(stdin, 'encoding') and stdin.encoding:
...
```

Can you try the code in dev and see if the problem is fixed?

Thanks.

**Comment by cannatag** [comment]
Fixed in release 2.3 of ldap3.

**Comment by cronicryo** [comment]
crap. i wish i saw your comment sooner. Just by looking at it it wouldnt fix the problem. hasattr is making sure that std.encoding doesnt run because the attribute doesnt exist which is throwing the error. 
in your modified version std.encoding is still running event though std doesnt have the encoding attribute
